cmd/main.go

```diff
@@ -68,6 +83,89 @@ func main() {
 	}
 
+	var cloud cloudPkg.Cloud
+	var k8sClient kubernetes.Interface
+	var md metadata.MetadataService
+
+	if _, ok := metadataRequiredModes[cmd]; ok {
+		options.Mode = driver.Mode(cmd)
+		options.AddFlags(fs)
+
+		if err := fs.Parse(args); err != nil {
+			klog.ErrorS(err, "Failed to parse options")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 0)
+		}
+		if err := options.Validate(); err != nil {
+			klog.ErrorS(err, "Invalid options")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 0)
+		}
+
+		err := logsapi.ValidateAndApply(c, featureGate)
+		if err != nil {
+			klog.ErrorS(err, "failed to validate and apply logging configuration")
+		}
+
+		cfg := metadata.MetadataServiceConfig{
+			MetadataSources: options.MetadataSources,
+			IMDSClient:      metadata.DefaultIMDSClient,
+			K8sAPIClient:    metadata.DefaultKubernetesAPIClient(options.Kubeconfig),
+		}
+
+		region := os.Getenv("AWS_REGION")
+		var metadataErr error
+
+		if region != "" {
+			klog.InfoS("Region provided via AWS_REGION environment variable", "region", region)
+			if options.Mode != driver.ControllerMode {
+				klog.InfoS("Node service requires metadata even if AWS_REGION provided, initializing metadata")
+				md, metadataErr = metadata.NewMetadataService(cfg, region)
+			}
+		} else {
+			klog.InfoS("Initializing metadata")
+			md, metadataErr = metadata.NewMetadataService(cfg, region)
+		}
+
+		if metadataErr != nil {
+			klog.ErrorS(metadataErr, "Failed to initialize metadata when it is required")
+			if options.Mode == driver.ControllerMode {
+				klog.InfoS("The region can be manually supplied via the AWS_REGION environment variable")
+			}
+			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+		} else if region == "" {
+			region = md.GetRegion()
+		}
+
+		var accountID string
+		if options.Mode == driver.ControllerMode || options.Mode == driver.AllMode {
+			cfg, err := config.LoadDefaultConfig(context.Background(), config.WithRegion(region))
+			if err != nil {
+				klog.ErrorS(err, "Failed to create AWS config for account ID retrieval")
+				klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+			}
+
+			stsClient := sts.NewFromConfig(cfg)
+			resp, err := stsClient.GetCallerIdentity(context.Background(), &sts.GetCallerIdentityInput{})
+			if err != nil {
+				klog.ErrorS(err, "Failed to get AWS account ID, HyperPod functionality may not work")
+				// Continue without account ID - existing functionality should still work
+			} else {
+				accountID = *resp.Account
+				klog.V(5).InfoS("Retrieved AWS account ID for HyperPod operations", "accountID", accountID)
+			}
+		}
+
+		cloud, err = cloudPkg.NewCloud(region, accountID, options.AwsSdkDebugLog, options.UserAgentExtra, options.Batching, options.DeprecatedMetrics)
+		if err != nil {
+			klog.ErrorS(err, "failed to create cloud service")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+		}
+
+		k8sClient, err = cfg.K8sAPIClient()
+		if err != nil {
+			klog.V(2).InfoS("Failed to setup k8s client", "err", err)
+		}
```

Comment:
Is it safe to have these empty in non-metadata cases? I don't think we did that before